### PR TITLE
🏗 Refine and update a few infrastructure related `OWNERS` files

### PR DESCRIPTION
**Pull request by rsimha**

### CODE REVIEW

**Review by kristoferbaxter — APPROVED** [review]
Only change I'm a little uncertain about is the `babel-config` owner change.

File: build-system/babel-config/OWNERS [hunk 1]
```diff
@@ -5,6 +5,7 @@
   rules: [
     {
       owners: [
+        {name: 'ampproject/wg-bento'},
```
Comment (jridgewell) [inline build-system/babel-config/OWNERS:8]:
```suggestion
```

File: build-system/compile/OWNERS [hunk 2]
```diff
@@ -9,6 +9,7 @@
     {
       pattern: '{bundles.config,sources}.js',
       owners: [
+        {name: 'ampproject/wg-bento'},
```
Comment (jridgewell) [inline build-system/compile/OWNERS:12]:
```suggestion
```

**Review by jridgewell — APPROVED** [review]
I don't think Bento WG is appropriate for these 2 dirs.

**Pull request merged**